front-end/js/jump.js

```diff
@@ -362,6 +392,171 @@ function registerJumps(viewId, svg, layerId) {
+                .html("Update Attributes")
+                .append("a")
+                .classed("view_" + viewId + " close", true)
+                .attr("href", "#")
+                .attr("id", "popoverclose")
+                .html("&times;")
+                .on("click", function() {
+                    removePopovers(viewId);
+                });
+            d3.select(viewClass + "#updatepopover")
+                .append("div")
+                .classed("view_" + viewId + " popover-content list-group", true)
+                .attr("id", "popovercontent");
+
+            // add attribute input boxes
+            let k;
+            for (k = 0; k < directMappedColNames.length; k++) {
+                let attrName = "<b>" + directMappedColNames[k] + "</b>";
+                let attrValue = directMappedColumns[directMappedColNames[k]];
+
+                let updateAttrs = d3
+                    .select(viewClass + "#popovercontent")
+                    .append("div")
+                    .classed("input-group mb-3", true)
+                    .attr("id", "attr-input-group-" + k);
+                
+                updateAttrs
+                    .append("div")
+                    .classed("input-group-prepend", true)
+                    .append("span")
+                    .classed("input-group-text", true)
+                    .attr("id", "inputGroup-sizing-default")
+                    .html(attrName);
+
+                updateAttrs
+                    .append("input")
+                    .classed("form-control attr-inputs", true)
+                    .attr("type", "text")
+                    .attr("value", attrValue)
+                    .attr("aria-label", "Default")
+                    .attr("aria-describedby", "inputGroup-sizing-default");
+            }
+
+            // add save changes button to bottom of popover
+            d3.select("#attr-input-group-" + (k-1))
+                .append("div")
+                .classed("popover-footer", true)
+                .append("button")
+                .classed("btn btn-success", true)
+                .attr("type", "button")
+                .attr("id", "update-button")
+                .html("Save");
+
+            // add listener to save changes button, sends updates to backend
+            d3.select("#update-button").on("click", function(d) {
+                d3.event.preventDefault();
+                let newAttrValues = [];
+                let objectKV = {};
+                d3.selectAll(".attr-inputs").each(function(d,i) {
+                    newAttrValues.push(d3.select(this).property("value"));
+                });
+
+                console.log(`column names -> ${directMappedColNames}`);
+                console.log(`new attr values -> ${newAttrValues}`);
+
+
+                for (let i = 0; i < directMappedColNames.length; i++) {
+                    let colName = directMappedColNames[i];
+                    objectKV[colName] = newAttrValues[i];
```

Comment:
nit pick: we can compare `directMappedColumns[colName]` with `newAttrValues[i]`. If they are the same, we don't need to update this field.